src/bin/ion/commands/beta/generate/templates/java/nested_type.templ

```diff
@@ -74,7 +76,11 @@
                                                             reader.{{ field.value_type | camel }}Value();
                                                        {% endif %}
                                                     {% else %}
-                                                        {{ field.value_type }}.readFrom(reader);
+                                                        {% if field.value_type is containing("ArrayList") %}
```

Comment:
It will. I will be changing the data model defined as per the current PR in `model.rs`. That has a reference to `CodeGenType` and based on this enum variant these checks can be defined properly.